### DISCUSSION TITLE
Make it easier to see what C# version is being used within a project.

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/roslyn#56213] [General] [answered]
It would be helpful if a developer could see at a glance which version of C# is active within a source file or project. This is currently only discoverable by looking it up on a webpage as a function of the .Net version which is rather contrived.

Perhaps simply display it in the project's properties:

![image](https://user-images.githubusercontent.com/12262952/132236707-45cfc2ca-159d-4562-8655-c0457aba6467.png)

Right now Language version is not displayed, just the text in the disabled dropdown `"Automatically selected based on framework version"` but it could be included in that text, e.g. `"C# 9.0 (Automatically selected based on framework version)"`.

It seems to me that even if a project targeted multiple frameworks then the lowest "wins", the choice of language version would/should be dictated by the lowest versioned framework.

In the IDE, when there are multiple frameworks we see the active one (so far as the IDE is concerned) as:

![image](https://user-images.githubusercontent.com/12262952/132245857-d9a9b961-e117-4f2f-81f8-a7bf98a89b41.png)

The text `(net5.0)` _could_ be enhanced to `(net5.0, C# 9.0)`.

**Comment by CyrusNajmabadi** [comment]
`#error version` is the best way to do this as it will tell you exactly this information and won't be affected by the ide tooling showing you sometime potentially different from what may end up running when msbuild executes.

**Reply by The-Futurist** [reply]
> `#error version` is the best way to do this as it will tell you exactly this information and won't be affected by the ide tooling showing you sometime potentially different from what may end up running when msbuild executes.

Not sure how you can claim "best" here. The supported C# version is dependent on framework version and the IDE clearly knows that.

**Comment by jnm2** [comment]
Here's what it looks like in VS2022, same thing. I agree, it would be nice to see `9.0` or `10.0 (preview)` stated here.

![image](https://user-images.githubusercontent.com/8040367/132245714-73544611-4606-4ac7-9e93-37e58ee42277.png)

I don't believe this question is related to the C# language design repo, though. I would guess at https://github.com/dotnet/project-system being the place to ask.

**Reply by CyrusNajmabadi** [reply]
@tmeschter do you know where this issue should move to?  It def shouldn't be at csharplang :-)

**Comment by The-Futurist** [comment]
I began by seeking to [put it here](https://github.com/dotnet/roslyn/issues/new/choose) but must have clicked on some odd link or other and ended up in csharplang without realizing.

**Comment by tmeschter** [comment, marked as answer]
@Korporal @CyrusNajmabadi @jnm2 You're in luck. This was added in https://github.com/dotnet/project-system/pull/7511 and will be available in C# SDK-style projects in VS 2022.

**Reply by The-Futurist** [reply]
@tmeschter - Hey, that's cool man, great, thanks!
